packages/core/src/factory.ts

```diff
@@ -66,7 +66,7 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
       components.mock({
         resource,
         input: { data, validations },
-        config: config.mock || {},
+        config: config.requestConfig || config.mock || {},
```

Comment:
should we blend these objects together?

Reply:
I went ahead and changed it so that proxy and mock both use the config.mock. Since we were using mock: false to determine if its a proxy or not, I went ahead and introduced isProxy to take its place.